lib/Hakyll/Web/Template/Context.hs

```diff
@@ -21,9 +21,13 @@
 --------------------------------------------------------------------------------
 {-# LANGUAGE CPP                       #-}
 {-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE FlexibleInstances #-}
 module Hakyll.Web.Template.Context
     ( ContextField (..)
     , Context (..)
+    , context
+    , functionContext
```

Comment:
I wonder whether we should export the general functions under new names. I don't really like `context` and `functionContext`, as they construct contexts for specific keys. Should be `keyContext` at best.
Do you think it would do any harm to just widen the type of `field` and `functionField`?

Reply:
I think it's fine to widen the type if it doesn't break existing code.  I'm a bit afraid it could break existing code though, since people usually use `{-# LANGUAGE OverloadedStrings #-}`.